### PR TITLE
Migrate to slog

### PR DESCRIPTION
**Pull request by majst01**
depends on https://github.com/metal-stack/metal-lib/pull/128

### CODE REVIEW

File: cmd/metal-api/internal/grpc/grpc-server.go [hunk 1]
```diff
@@ -72,56 +75,84 @@ func Run(cfg *ServerConfig) error {
...
-			log.Errorf("[PANIC] %s stack:%s", p, string(debug.Stack()))
-			return status.Errorf(codes.Internal, "%s", p)
-		},
-	)
+	// recoveryOpt := grpc_recovery.WithRecoveryHandlerContext(
+	// 	func(ctx context.Context, p any) error {
+	// 		log.Error("[PANIC] %s stack:%s", p, string(debug.Stack()))
+	// 		return status.Errorf(codes.Internal, "%s", p)
+	// 	},
+	// )
```
Comment (mwennrich) [inline cmd/metal-api/internal/grpc/grpc-server.go:86]:
```suggestion
	recoveryOpt := recovery.WithRecoveryHandlerContext(
		func(ctx context.Context, p any) error {
			log.Error("[PANIC] %s stack:%s", p, string(debug.Stack()))
			return status.Errorf(codes.Internal, "%s", p)
		},
	)
```

File: cmd/metal-api/internal/grpc/grpc-server.go [hunk 2]
```diff
@@ -72,56 +75,84 @@ func Run(cfg *ServerConfig) error {
...
+		}
+		auditUnaryInterceptor, err := auditing.UnaryServerInterceptor(cfg.Auditing, log.WithGroup("auditing-grpc"), shouldAudit)
+		if err != nil {
+			return err
+		}
+		streamInterceptors = append(streamInterceptors, auditStreamInterceptor)
+		unaryInterceptors = append(unaryInterceptors, auditUnaryInterceptor)
 	}
 
 	unaryInterceptors = append(unaryInterceptors, metrics.GrpcMetrics)
```
Comment (mwennrich) [inline cmd/metal-api/internal/grpc/grpc-server.go:146]:
```suggestion
	unaryInterceptors = append(unaryInterceptors,recovery.UnaryServerInterceptor(recoveryOpt))
	streamInterceptors = append(streamInterceptors, recovery.StreamServerInterceptor(recoveryOpt))

```

**Review by mwennrich — CHANGES_REQUESTED** [review]

File: cmd/metal-api/internal/grpc/grpc-server.go (hunk 2 above)
Comment (majst01) [reply cmd/metal-api/internal/grpc/grpc-server.go:145]: Nice, thanks, done

File: cmd/metal-api/internal/grpc/grpc-server.go (hunk 1 above)
Comment (majst01) [reply cmd/metal-api/internal/grpc/grpc-server.go]: Nice, thanks, done

File: cmd/metal-api/internal/eventbus/nsq.go [hunk 3]
```diff
@@ -37,18 +37,18 @@ func (n *NSQClient) WaitForPublisher() {
...
-		n.logger.Sugar().Infow("nsq connected", "nsqd", fmt.Sprintf("%+v", n.config))
+		n.logger.Info("nsq connected", "nsqd", fmt.Sprintf("%+v", n.config))
 		n.Publisher = publisher
 		break
 	}
 }
 
 func (n *NSQClient) CreateEndpoints(lookupds ...string) error {
-	c, err := bus.NewConsumer(n.logger, n.config.TLS, lookupds...)
+	c, err := bus.NewConsumer(n.logger, n.config.TLS, lookupds...) // FIXME
```
Comment (Gerrit91) [inline cmd/metal-api/internal/eventbus/nsq.go:51]: What needs to be fixed?

File: cmd/metal-api/internal/service/machine-service_allocation_test.go [hunk 4]
```diff
@@ -335,7 +335,7 @@ func setupTestEnvironment(machineCount int, t *testing.T) (*datastore.RethinkSto
 	ms, err := NewMachine(log, rs, &emptyPublisher{}, bus.DirectEndpoints(), ipam.New(ipamer), mdc, nil, usergetter, 0, nil, metal.DisabledIPMISuperUser())
 	require.NoError(t, err)
 	container := restful.NewContainer().Add(ms)
-	container.Filter(rest.UserAuth(usergetter, zaptest.NewLogger(t).Sugar()))
+	container.Filter(rest.UserAuth(usergetter, slog.Default())) // FIXME
```
Comment (Gerrit91) [inline cmd/metal-api/internal/service/machine-service_allocation_test.go:338]: What needs to be fixed?

File: cmd/metal-api/internal/eventbus/nsq.go (hunk 3 above)
Comment (majst01) [reply cmd/metal-api/internal/eventbus/nsq.go]: nothing, done

**Review by mwennrich — APPROVED** [review]

**Review by Gerrit91 — APPROVED** [review]

**Pull request merged**